pkg/resources/services/service.go

```diff
@@ -20,13 +20,27 @@ package services
 import (
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
 
 	miniov1 "github.com/minio/operator/pkg/apis/minio.min.io/v1"
 )
 
 // NewClusterIPForMinIO will return a new headless Kubernetes service for a Tenant
 func NewClusterIPForMinIO(t *miniov1.Tenant) *corev1.Service {
-	minioPort := corev1.ServicePort{Port: miniov1.MinIOPort, Name: miniov1.MinIOServicePortName}
+	var port int32
+	var name string
+	if t.TLS() {
+		port = 443
+		name = miniov1.MinIOServiceHTTPSPortName
+	} else {
+		port = 80
+		name = miniov1.MinIOServiceHTTPPortName
+	}
+	minioPort := corev1.ServicePort{
+		Port:       port,
+		Name:       name,
+		TargetPort: intstr.FromInt(miniov1.MinIOPort),
```

Comment:
why not use strconv.Itoa() ?

Reply:
Using the API framework to keep the types the same: https://godoc.org/k8s.io/api/core/v1#ServicePort

Reply:
Wow, that is so ugly!